distributed/scheduler.py

```diff
@@ -4860,6 +4866,7 @@ def _generate_taskstates(
                     # dask/dask#9888.
                     dependencies[k] = deps_lhs
 
+                    colliding_tasks.append(ts)
```

Comment:
nit: not relevant for this specific instance but this entire code path is rather performance sensitive both in terms of CPU but also memory usage. If we don't need to track the tasks, you should just use a counter here because later you'll just take the length of this list anyhow.

This also reduces complexity since refactoring this code is a pain

Reply:
Makes sense, I wasn't sure if I wanted to do more than just collect the size but didn't end up doing anything else.